File: Web/src/test/java/forms/FirstCardForm.java
package forms;

import aquality.selenium.elements.interfaces.*;
import aquality.selenium.forms.Form;
import org.openqa.selenium.By;
import utils.WorkWithDomain;

public class FirstCardForm extends Form {

    private final ITextBox passwordInput = getElementFactory().getTextBox(By.xpath("//input[@placeholder='Choose Password']"),"password input");
    private final ITextBox emailInput = getElementFactory().getTextBox(By.xpath("//input[@placeholder='Your email']"),"email input");
    private final ITextBox domainInput = getElementFactory().getTextBox(By.xpath("//input[@placeholder='Domain']"),"domain input");
    private final IComboBox otherSelectCB = getElementFactory().getComboBox(By.xpath("//div[@class='dropdown__field']"),"other select CB");
    private final ICheckBox licenseCB = getElementFactory().getCheckBox(By.xpath("//span[@class='icon icon-check checkbox__check']"),"acceptLicence");
    private final ILink goNextCard = getElementFactory().getLink(By.xpath("//a[@class='button--secondary']"),"go next card");

    public FirstCardForm() {
        super(By.xpath("//div[@class='login-form__container']"), "loginCard");
    }

    public void clearPasswordFieldAndType(String password){
        passwordInput.clearAndType(password);
    }

    public void clearEmailFieldAndType(String email){
        emailInput.clearAndType(email);
    }

    public void clearDomainAndTYpe(String damain){
        domainInput.clearAndType(damain);
    }

    public void otherSelect(){
        otherSelectCB.click();
    }

    public void otherValueSelect(String domainValue){

        WorkWithDomain.getDomainLabel(domainValue).click();
    }

    public void useLicenseCheckBox(){
        licenseCB.click();
    }

    public void goNextCard(){
        goNextCard.click();
    }
}

File: Web/UserInterfaceTest/UserInterfaceTest/Forms/Pages/Cards/FirstCardForm.cs
using OpenQA.Selenium;
using Aquality.Selenium.Elements.Interfaces;
using UserInterfaceTest.Models;


namespace UserInterfaceTest.Forms.Pages.Cards
{
    public class FirstCardForm : BaseAppForm
    {
        private ITextBox PasswordInput => ElementFactory.GetTextBox(By.XPath("//input[contains(@placeholder,'Choose Password')]"), "Password input");
        private ITextBox EmailLoginInput => ElementFactory.GetTextBox(By.XPath("//input[contains(@placeholder,'Your email')]"), "Email login input");
        private ITextBox EmailServerInput => ElementFactory.GetTextBox(By.XPath("//input[contains(@placeholder,'Domain')]"), "Email server input");
        private ILabel EmailDomain => ElementFactory.GetLabel(By.XPath("//div[@class='dropdown__field']"),"domain input");
        private ICheckBox AcceptTermsAndConditions => ElementFactory.GetCheckBox(By.XPath("//label[@class='checkbox__label']"),
                                                                                       "Accept terms andc onditions checkbox");
        private IButton NextCard => ElementFactory.GetButton(By.XPath("//a[contains(@class,'button--secondary')]"),"move to 2 card button");
        private string ItemXPathTemplate = "//div[contains(@class,'list-item') and text() = '{0}']";
        
      

        public FirstCardForm() : base(By.XPath("//input[contains(@placeholder,'Choose Password')]"), "Password input")
        {
        }

        public void SetUser(User user)
        {
            PasswordInput.ClearAndType(user.Password);
            EmailLoginInput.ClearAndType(user.EmailLogin);
            EmailServerInput.ClearAndType(user.Server);
            EmailDomain.Click();
            var item = ElementFactory.GetLabel(By.XPath(string.Format(ItemXPathTemplate,user.Domain)),"domain item");
            item.Click();
        }

        public void ClickOnTermsCheckBox() => AcceptTermsAndConditions.Click();
        public void ClickNext()=> NextCard.Click();

    }
}
